generator/types.go

```diff
@@ -368,11 +401,30 @@ func (m *Model) CtorRetVars() string {
 }
 
 // SetFields sets all the children fields and their model to the current model.
+// It also finds the primary key and sets it in the model.
 func (m *Model) SetFields(fields []*Field) {
+	var fs []*Field
+	var id *Field
 	for _, f := range fields {
+		if f.Type == BaseModel {
+			m.ID = f.Fields[0]
+			m.ID.Type = defaultIDType
+			id = f
+		}
+
 		f.Model = m
-		m.Fields = append(m.Fields, f)
+		if f.IsPrimaryKey() {
+			id = f
```

Comment:
Is it guarantied in any other place that there is only one field that f.IsPrimaryKey() == true?
If we are not supported compounde PK, we should maybe "panic" if the user tries to define a compounded one